src/lib.rs

```diff
@@ -218,9 +218,23 @@ impl<T: ContactInfo> RoutingTable<T> {
                 } else {
                     vec![]
                 };
+
                 self.buckets[bucket_index].insert(i, info);
+
+                let not_needed = if self.buckets[bucket_index].len() > GROUP_SIZE {
+                    self.buckets[bucket_index]
+                        .iter()
+                        .rev()
+                        .take(self.buckets[bucket_index].len() - GROUP_SIZE)
+                        .cloned()
+                        .collect()
```

Comment:
You probably don't even need that `if`: Just `self.buckets[bucket_index].iter().skip(GROUP_SIZE).cloned().collect()` should do it.